test/unit/org/apache/cassandra/service/accord/journal/AccordTopologyUpdateTest.java

```diff
@@ -0,0 +1,175 @@
+{
+    private static final long[] EPOCHS = new long[0];
+    private static final Ranges[] RANGES = new Ranges[0];
+    private static final TableId TBL1 = TableId.fromRaw(0, 0);
+
+    static
+    {
+        DatabaseDescriptor.clientInitialization();
+    }
+
+    @Before
+    public void before()
+    {
+        DatabaseDescriptor.setPartitionerUnsafe(Murmur3Partitioner.instance);
+    }
+
+    @Test
+    public void rangesForEpoch()
+    {
+        @SuppressWarnings({ "resource", "IOResourceOpenedButNotSafelyClosed" }) DataOutputBuffer output = new DataOutputBuffer();
+        qt().forAll(rangesForEpochGen()).check(expected -> {
+            maybeUpdatePartitioner(expected);
+            Serializers.testSerde(output, AccordTopologyUpdate.RangesForEpochSerializer.instance, expected);
+        });
+    }
+
+    @Test
+    public void topologyUpdate()
+    {
+        @SuppressWarnings({ "resource", "IOResourceOpenedButNotSafelyClosed" }) DataOutputBuffer output = new DataOutputBuffer();
+        qt().forAll(topologyUpdateGen()).check(expected -> {
+            maybeUpdatePartitioner(expected);
+            Serializers.testSerde(output, AccordTopologyUpdate.TopologyUpdateSerializer.instance, expected);
+        });
+    }
+
+    @Test
+    public void accordTopologyUpdate()
+    {
+        @SuppressWarnings({ "resource", "IOResourceOpenedButNotSafelyClosed" }) DataOutputBuffer output = new DataOutputBuffer();
+        qt().forAll(accordTopologyUpdateGen()).check(expected -> {
+            maybeUpdatePartitioner(expected);
+            Serializers.testSerde(output, AccordTopologyUpdate.Serializer.instance, expected);
+        });
+    }
+
+    private static Gen<CommandStores.RangesForEpoch> rangesForEpochGen()
+    {
+        return AccordGenerators.partitioner().flatMap(p -> rangesForEpochGen(AccordGenerators.rangesSplitOrArbitrary(p)));
+    }
+
+    private static Gen<CommandStores.RangesForEpoch> rangesForEpochGen(Gen<Ranges> rangesGen)
+    {
+        Gen.IntGen sizeGen = Gens.ints().between(0, 10);
+        Gen.LongGen epochGen = AccordGens.epochs();
+        return rs -> {
+            int size = sizeGen.nextInt(rs);
+            if (size == 0)
+                return new CommandStores.RangesForEpoch(EPOCHS, RANGES);
+            long epoch = epochGen.nextLong(rs);
+            long[] epochs = new long[size];
+            Ranges[] ranges = new Ranges[size];
+            for (int i = 0; i < size; i++)
+            {
+                epochs[i] = epoch++;
+                ranges[i] = rangesGen.next(rs);
+            }
+            return new CommandStores.RangesForEpoch(epochs, ranges);
+        };
+    }
+
+    private static Gen<Journal.TopologyUpdate> topologyUpdateGen()
+    {
+        Gen<IPartitioner> partitionerGen = AccordGenerators.partitioner();
+        return rs -> {
+            IPartitioner partitioner = partitionerGen.next(rs);
+            Gen<Ranges> rangesGen = AccordGenerators.ranges(TBL1, partitioner);
```

Comment:
limit to a single table as the perf of this generator isn't the best, we spend a lot of time creating ranges